OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/EditorTopComponent.java

```diff
@@ -276,6 +287,51 @@ public void commentOutText() {
         }
 
     }
+    
+    public void spellCheckAllText() throws BadLocationException {     
+        Document doc = rSyntaxTextArea.getDocument();   
+        if (doc != null) {                                                
+            String editorText = rSyntaxTextArea.getText(0, doc.getLength());            
+            if (editorText != null) {  
+                Highlighter highlighter = rSyntaxTextArea.getHighlighter();
+                Highlighter.HighlightPainter painter = new DefaultHighlighter.DefaultHighlightPainter(Color.PINK);  //Default color is: PINK
+
+                JLanguageTool langTool = new JLanguageTool(new AmericanEnglish());  //Default language is: American English
+                for (Rule rule : langTool.getAllActiveRules()) {
+                    if (rule instanceof SpellingCheckRule) {
+                        ((SpellingCheckRule)rule).acceptPhrases(Arrays.asList("tex", "Tex", "TEX", "documentclass", "maketitle"));  //Accept some TEX terms
+                    }
+                }

```

Comment:
The LanguageTool object and all the rules should not be getting created/configured each time I click the "check" button. Ideally this should live in a separate class and be initiated at the application startup. During runtime, we would only call "check" on the tool. I think this is the main reason the app hangs for some time after clicking "check"